pypesto/visualize/optimize.py

```diff
@@ -0,0 +1,70 @@
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import matplotlib.cm as cm
+from matplotlib.ticker import MaxNLocator
+from scipy import cluster
+import numpy as np
+
+
+def waterfall(result, ax=None):
+    """
+    Plot waterfall plot.
+
+    Parameters
+    ----------
+
+    result: pypesto.Result
+        Optimization result obtained by 'optimize.py'
+
+    ax: matplotlib.Axes, optional
+        Axes object to use.
+
+    Returns
+    -------
+
+    ax: matplotlib.Axes
+        The plot axes.
+    """
+
+    # axes
+    if ax is None:
+        fig, ax = plt.subplots()
+
+    # extract cost function values from result
+    result_fval = result.optimize_result.get_for_key('fval')
+    result_fval = np.reshape(result_fval, [len(result_fval), 1])
+    startpoints = range(1, len(result_fval) + 1)
+
+    # cluster
+    clust = cluster.hierarchy.fcluster(
```

Comment:
Move clustering to separate function for readability and reuse. This will be used for parameter plots as well.

Reply:
Consider adding a test for the clustering, including vectors containing NaN and Inf.